web/src/components/settings-page/project/components/rbac-form/components/role/index.tsx

```diff
@@ -34,58 +43,84 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 const SUB_SECTION_TITLE = "Role";
-const RESOURCE_ACTION_SEPARATOR = ";";
-const RESOURCES_KEY = "resources";
-const ACTIONS_KEY = "actions";
 
 export const RoleTable: FC = memo(function RoleTable() {
   const classes = useStyles();
   const projectSettingClasses = useProjectSettingStyles();
   const rbacRoles = useAppSelector((state) => state.project.rbacRoles);
   const dispatch = useAppDispatch();
-  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(
-    null
-  );
+  const [isOpenAddForm, setIsOpenAddForm] = useState(false);
+  const [deleteRole, setDeleteRole] = useState<null | string>(null);
+  const [editRole, setEditRole] = useState<null | string>(null);
 
-  const formalizePoliciesFromPoliciesList = ({
-    policiesList,
-  }: {
-    policiesList: Array<RBACPolicy>;
-  }): Array<string> => {
-    const policies: Array<string> = [];
-    policiesList.map((policy) => {
-      const resources: Array<string> = [];
-      policy.resourcesList.map((resource) => {
-        resources.push(RBAC_RESOURCE_TYPE_TEXT[resource.type]);
+  const handleSubmit = useCallback(
+    (values: { name: string; policies: string }) => {
+      const params = {
+        name: values.name,
+        policies: parseRBACPolicies({ policies: values.policies }),
+      };
+      dispatch(addRBACRole(params)).then((result) => {
+        if (addRBACRole.fulfilled.match(result)) {
```

Comment:
I'm still a bit worried about this one. Let's say, this `then` will run after the dispatch function finish, then if the API handling is still not finished, it means that if the check failed, and even if the API runs successfully, the toast and so on handlers will not be triggered. This way to handle only ensures that there will be no error toast and success toast showing at the same time, but in other cases problem existed, wdyt?

Reply:
lets check and handle that later since we don't have plan for this currently 🤔

Reply:
I see, thanks for reviewing.